MediumProject/blog/views.py

```diff
@@ -0,0 +1,120 @@
+        user = serializer.save()
+
+        return Response({
+            "user": UserSerializer(user, context=self.get_serializer_context()).data,
+            "token": AuthToken.objects.create(user)[1]
+        })
+
+
+class LoginView(generics.GenericAPIView):
+    serializer_class = LoginUserSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = serializer.validated_data
+
+        return Response({
+            "user": UserSerializer(user, context=self.get_serializer_context()).data,
+            "token": AuthToken.objects.create(user)[1]
+        })
+
+
+class PostListCreateView(generics.ListCreateAPIView):
+    queryset = Post.objects.all()
+    serializer_class = PostListSerializer
+    permission_classes = [IsAuthenticatedOrReadOnly]
+    pagination_class = CursorPagination
+
+    def get(self, request, *args, **kwargs):
+        print(request.user)
+        return super().get(request, *args, **kwargs)
+
+    def permission_denied(self, request, message=None, code=None):
+        raise PermissionDenied(message)
+
+
+class PostRetrieveUpdateDestroyView(generics.RetrieveUpdateDestroyAPIView):
+    permission_classes = [IsCreatorOrReadOnly | IsAdminUser]
+    queryset = Post.objects.all()
+
+    def get_serializer_class(self):
+        if self.request.method == 'GET':
+            return PostDetailSerializer
+        return PostListSerializer
+
+    def permission_denied(self, request, message=None, code=None):
+        raise PermissionDenied(message)
+
+
+class CommentListCreateView(generics.ListCreateAPIView):
+    serializer_class = CommentListSerializer
+    permission_classes = [IsAuthenticatedOrReadOnly]
+    pagination_class = CursorPagination
+
+    def get_queryset(self):
+        return Comment.objects.filter(post=self.kwargs['pk'])
+
+    def permission_denied(self, request, message=None, code=None):
+        raise PermissionDenied(message)
+
+
+class CommentRetrieveUpdateDestroyView(generics.UpdateAPIView, generics.DestroyAPIView, GenericAPIView):
+    permission_classes = [IsCreatorOrReadOnly | IsAdminUser]
+    serializer_class = CommentListSerializer
+    queryset = Comment.objects.all()
+
+    def get_object(self):
+        obj = Comment.objects.get(pk=self.kwargs['id'])
+        self.check_object_permissions(self.request, obj)
+        return obj
+
+    def check_object_permissions(self, request, obj):
+        super().check_object_permissions(request, obj)
+
+    def permission_denied(self, request, message=None, code=None):
+        raise PermissionDenied(message)
+
+    def put(self, request, *args, **kwargs):
+        return self.update(request, *args, **kwargs)
+
+    def patch(self, request, *args, **kwargs):
+        return self.partial_update(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        return self.destroy(request, *args, **kwargs)
```

Comment:
요기는 테스트용으로 남아있는 코드일까요?

Reply:
아 요기가 질문 2 부분이군요. mixin이 아닌 Views를 상속하셔서 생기는 문제인 것 같습니다. 

요런 식으로 수정하시면 의도한 바대로 수정하실 수 있을 것 같아요~

```suggestion python
class CommentRetrieveUpdateDestroyView(UpdateModelMixin, DestroyModelMixin, GenericAPIView):
```

Reply:
제가 처음에 이런 방식으로 했었는데 요청을 보내니 put, patch, delete Method가 다 허용되지 않은 method라고 떴습니다..! 문제의 원인을 찾을 수가 없어 일단 위와 같이 해결했습니다 :/